### ISSUE TITLE
Image paste from clipboard broken

### ISSUE THREAD

**Issue by Ben-MyBB** [mybb/mybb#2440]
Hi,

just installed myBB on our server. First thing I tested was image paste from the clipboard ... but this seems to be broken. Is this a known issue? I couldn't find it in the forum or on github.
Short description:
Paste image works in the editor, but displays as source code in the preview and in the final posted version

If its not a known issue, I'm happy to provide more details.

Jan

Original thread: [Image paste from clipboard broken](http://community.mybb.com/thread-194877.html)

**Comment by Destroy666x** [comment]
Not a bug - lack of a commonly used feature supported by many scripts (including SCEditor, ofc temporarily) rather as I mentioned on Slack not so long ago. Would be great to have, I'm most likely going to work on it after my exams if noone else does.

**Comment by effone** [comment]
Its not related to SCEditor. The editor works fine.
Its MyBB parser which is not familiar with data uri.
A separate PR is desired targeting MyBB Parser only ...

**Comment by effone** [comment]
So, as I have stated, after a double confirmation:
https://github.com/mybb/mybb/pull/3639
This update fixes issue related to paste image from clipboard. Its now working fine and showing proper preview at frontend. But as MyBB Parser is not familiar with `data-uri` the image gets broken after posting the message.

**Comment by euantorano** [comment]
As I said in the other issue, I’d be happy to add data URI support to the parser.

**Comment by sencagri** [comment]
We still can't put images from clipboard. Is there any improvoment ?

**Comment by euantorano** [comment]
We reverted a patch implementing support as it turned out to cause problems. Adding support for it is going to require some further work.

**Comment by JordanMussi** [comment]
@euantorano do you want to leave this issue open to track the need to upload pastes that are files instead of URLs? Since the fix for #4078 is to use `[img]<URL>[/img]` where possible and any pastes eg from copying a local file will paste nothing into the editor.

**Comment by euantorano** [comment]
Yeah, I think that's for the best @JordanMussi.

**Comment by AlbertHart** [comment]
Of course, this would be a great feature - the ability to paste a clipboard image into a post - have it automatically uploaded to the server, or placed in the MySQL database, and displayed on the page.

Many, many other sites and forums do this  - including this GitHUB issues forum.

If users are worried about disk space, they could easily disable this feature - in which case an error alert could appear when you try to paste, (or drop), an image into a message.

Here is a small image from my clipboard which I just pasted into this post on GitHUB.
(Boy, is this a great feature of GitHUB which I use in almost all of my posts - because a picture with arrows and such is such a great way to explain an issue)

![image](https://user-images.githubusercontent.com/8965352/86205549-63296f80-bb27-11ea-9312-894e752a6edb.png)
